objects/src/accounts/mod.rs

```diff
@@ -327,20 +327,22 @@ pub mod testing {
         Account::new(id, vault, storage, code, nonce)
     }
 
-    pub fn build_account_delta(
-        added_assets: Vec<Asset>,
-        removed_assets: Vec<Asset>,
-        nonce: Felt,
-    ) -> AccountDelta {
+    pub fn build_storage_delta() -> AccountStorageDelta {
         let word = [Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)];
-        let storage_delta = AccountStorageDelta {
+        AccountStorageDelta {
             cleared_items: vec![0],
             updated_items: vec![(1, word)],
             updated_maps: vec![],
-        };
+        }
+    }
 
+    pub fn build_account_delta(
+        added_assets: Vec<Asset>,
+        removed_assets: Vec<Asset>,
+        nonce: Felt,
+        storage_delta: AccountStorageDelta,
+    ) -> AccountDelta {
```

Comment:
One of the main intents for making `build_account_delta()` more general is so that we can use it in tests below to test whether `apply_delta()` works correctly. Right now, all `apply_delta()` tests only tests the case when we have the hard-coded storage delta (previously inside `build_account_delta()` and now inside `build_storage_delta()`.

Maybe the better approach here, as you've suggested, would be to create `AccountDeltaBuilder` which could go into `src/accounts/delta/builder.rs` module. For now, we'd use it only for testing (in the tests below). In the future, we could use it for other things too.

Reply:
Implemented the builder